packages/react/src/dialog/portal/DialogPortalContext.ts

```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+
+export const DialogPortalContext = React.createContext(false);
+
+export function useThrowIfPortalIsMissing() {
+  const value = React.useContext(DialogPortalContext);
+  if (!value) {
+    throw new Error('Base UI: <AlertDialog.Portal> is missing.');
+  }
+}
```

Comment:
Throws inside each `Popup` part with the name of the component that's missing a `Portal` part